msgpack-core/src/main/java/org/msgpack/value/ValueFactory.java

```diff
@@ -87,12 +87,32 @@ public static ImmutableFloatValue newFloat(double v)
 
     public static ImmutableBinaryValue newBinary(byte[] b)
     {
-        return new ImmutableBinaryValueImpl(b);
+        return newBinary(b, false);
+    }
+
+    public static ImmutableBinaryValue newBinary(byte[] b, boolean ref)
```

Comment:
The "correct" name is `useReferenceIfPossible` but it seems too long. How about `boolean omitCopy` instead?
I think inverting true/false is confusing because using true as the default seems unexpected.